### PR TITLE
HC: collect commitments

### LINKED ISSUE #3930: HC: collect commitments
Collect all commitments from the parent chain. Filter out _correct_ ones and feed them to the smart contract. This would yeld the next leader.

This depends on #3928 and #3929

### PR DESCRIPTION
**Pull request by velzevur**
Introduces commitments fetching from the parent chain and feeding it to the
election smart contract.

Closes #3930

The work on this PR had been supported by ACF.

Edit: I still need to fix the function names in the logs in `sec_consensus_hc`...

### CODE REVIEW

**Review by seanhinde — DISMISSED** [review]

File: apps/aehttp/test/aehttp_parent_connector_tests.erl [hunk 1]
```diff
@@ -96,17 +107,20 @@ ae_sim_test_() ->
                             aec_chain_sim:add_keyblock(SimName)
                         end,
                     %% post a different amount of blocks on different sims
-                    {ok, KB1} = PostPCBlock(Spec1),
+                    {ok, KB2} = PostPCBlock(Spec1),
 
-                    {ok, KB1} = PostPCBlock(Spec2),
                     {ok, KB2} = PostPCBlock(Spec2),
+                    {ok, KBr} = PostPCBlock(Spec2),
```
Comment (uwiger) [inline apps/aehttp/test/aehttp_parent_connector_tests.erl:113]: Is `KBr`  used anywhere?

File: test/contracts/HCElection.aes [hunk 2]
```diff
@@ -56,3 +69,9 @@ main contract HCElection =
   function assert_protocol_call() =
       require(Call.caller == Contract.creator, "Must be called by the protocol")
 
+  function filter_posted_commitmens((accum, total_s, commitments), (addr, stake)) =
```
Comment (uwiger) [inline test/contracts/HCElection.aes:72]:
```suggestion
  function filter_posted_commitments((accum, total_s, commitments), (addr, stake)) =
```

File: test/contracts/HCElection.aes [hunk 3]
```diff
@@ -23,16 +25,27 @@ main contract HCElection =
...
     let entropy : hash = Crypto.sha256(entropy_str)
-    let sorted = state.main_staking_ct.sorted_validators() 
-    let shot = Bytes.to_int(entropy) * height mod state.main_staking_ct.total_stake()
+    let sorted0 = state.main_staking_ct.sorted_validators() 
+    let (sorted, total_stake) =
+        if (Chain.block_height == 0)
+            (sorted0, state.main_staking_ct.total_stake())
+        else
+            let accepted_commitments = commitments[current_hash]
+            let (sorted1, total_s, _)  = List.foldl(filter_posted_commitmens, ([], 0, accepted_commitments), sorted0)
```
Comment (uwiger) [inline test/contracts/HCElection.aes:46]:
```suggestion
            let (sorted1, total_s, _)  = List.foldl(filter_posted_commitments, ([], 0, accepted_commitments), sorted0)
```

**Review by uwiger — APPROVED** [review]
LGTM

**Review by uwiger — APPROVED** [review]

**Review by seanhinde — DISMISSED** [review]

**Review by seanhinde — APPROVED** [review]

**Pull request merged**